infrastructure/templates/public-api/components/functionApp.bicep

```diff
@@ -68,67 +95,63 @@ resource appServicePlan 'Microsoft.Web/serverfarms@2023-01-01' = {
   }
 }
 
-var dedicatedStorageAccountName = replace('${subscription}eessa${functionAppName}', '-', '')
+// Configuring a single shared storage account for task management, and 2 individual storage accounts to be split
+// between the production slot and staging slot. See
+// https://learn.microsoft.com/en-us/azure/azure-functions/durable/durable-functions-zero-downtime-deployment#status-check-with-slot
+var durableManagementStorageAccountName = replace('${subscription}eessa${functionAppName}mg', '-', '')
+var slot1StorageAccountName = replace('${subscription}eessa${functionAppName}s1', '-', '')
+var slot2StorageAccountName = replace('${subscription}eessa${functionAppName}s2', '-', '')
 
 // Create a dedicated Storage Account for this Function App for it to store its deployment code and jobs in.
 // Grant the Function App access by whitelisting its subnet for inbound traffic.
 //
 // For performance, it is considered good practice for each Function App to have its own dedicated Storage Account. See
 // https://learn.microsoft.com/en-us/azure/azure-functions/storage-considerations?tabs=azure-cli#optimize-storage-performance.
-resource dedicatedStorageAccount 'Microsoft.Storage/storageAccounts@2023-01-01' = {
-  name: dedicatedStorageAccountName
-  location: location
-  sku: {
-    name: 'Standard_LRS'
-  }
-  kind: 'Storage'
-  properties: {
-    supportsHttpsTrafficOnly: true
-    defaultToOAuthAuthentication: true
-    networkAcls: {
-      bypass: 'AzureServices'
-      defaultAction: 'Deny'
-      virtualNetworkRules: [
-        {
-          action: 'Allow'
-          id: subnetId
-        }
-      ]
-    }
+
+// TODO EES-5128 - add private endpoints to allow VNet traffic to go directly to Storage Account over the VNet.
+module durableManagementStorageAccountModule 'storageAccount.bicep' = {
+  name: '${durableManagementStorageAccountName}StorageAccountDeploy'
+  params: {
+    location: location
+    storageAccountName: durableManagementStorageAccountName
+    allowedSubnetIds: [subnetId]
+    skuStorageResource: 'Standard_LRS'
+    keyVaultName: keyVaultName
+    tagValues: tagValues
   }
 }
```

Comment:
Related to the comment above, shouldn't this be named something like `sharedStorageAccountName` as its used to configure the `AzureWebJobsStorage` connection string?

Reply:
:+1: done ta